File: src/ca/qc/bdeb/inf203/model/combatants/VeggieKamikaz.java
package ca.qc.bdeb.inf203.model.combatants;

import ca.qc.bdeb.inf203.model.Action;
import ca.qc.bdeb.inf203.model.Combattant;
import ca.qc.bdeb.inf203.model.Entite;
import ca.qc.bdeb.inf203.model.Etat;
import ca.qc.bdeb.inf203.model.RepresentationImage;

/**
 * Veggie
 *
 * @author Guillaume Riou, Nicolas Hurtubise
 */
public class VeggieKamikaz extends Combattant {

    public VeggieKamikaz() {
        super();
        initialise();
    }

    @Override
    protected void initialise() {
        super.initialise();

        this.attaque = 100;
        this.gentil = false;
        this.hitbox.height = 80;
        this.hitbox.width = 60;
        this.animationFrameRate = 5;

        this.vie = 20;

        this.nbrImagesAnimation.put(Etat.DEPLACEMENT, 5);
        this.nbrImagesAnimation.put(Etat.ATTAQUE, 7);

        this.vitesseAction.put(Action.DEPLACEMENT, -116f);
        this.vitesseAction.put(Action.ATTAQUE, 0.6f);

        this.sprites.put(Etat.DEPLACEMENT, new RepresentationImage(new String[]{"veggies", "kamikaz", "walking"}));
        this.sprites.put(Etat.ATTAQUE, new RepresentationImage(new String[]{"explosion"}));
    }
    /**
     * Meurt après avoir attaqué.
     * @param nbrFois 
     */
    @Override
    protected void attaquer(int nbrFois) {
        super.attaquer(nbrFois);
        if (nbrFois > 0) {
            this.setVie(0);
        }
    }

    @Override
    public Entite action(int nbFois) {
        return null;
    }
}

File: assets/js/model/combatants/veggie-kamikaz.js
/**
 * Veggie kamikaz
 *
 * @author Guillaume Riou, Nicolas Hurtubise
 */
function VeggieKamikaz() {
    Combattant.apply(this);
    this.initialise();
}

VeggieKamikaz.prototype = new Combattant();

VeggieKamikaz.prototype.initialise = function() {
    this.attaque = 100;
    this.gentil = false;
    this.hitbox.height = 80;
    this.hitbox.width = 60;
    this.animationFrameRate = 5;

    this.vie = 20;

    this.nbrImagesAnimation[navigator.Etat.DEPLACEMENT] = 5;
    this.nbrImagesAnimation[navigator.Etat.ATTAQUE] = 7;

    this.vitesseAction[navigator.Action.DEPLACEMENT] = -116;
    this.vitesseAction[navigator.Action.ATTAQUE] = 0.6;

    this.sprites[navigator.Etat.DEPLACEMENT] = new RepresentationImage(["veggies", "kamikaz", "walking"]);
    this.sprites[navigator.Etat.ATTAQUE] = new RepresentationImage(["explosion"]);
};

/**
 * Meurt après avoir attaqué.
 * @param nbrFois 
 */
VeggieKamikaz.prototype.attaquer = function(nbrFois) {
    // Entite[]
    var aEnlever = [];
    for (var j = 0; j < nbFois; j++) {

        for (var index in this.cibles) {
            var cible = this.cibles[index];

            if (this.isEnnemi(cible)) {
                cible.vie -= this.attaque;
            } else {
                aEnlever.push(cible);
            }

        }
    }

    for (var index in this.cibles) {
        var cible = this.cibles[index];
        if (cible.vie <= 0) {
            aEnlever.push(cible);
        }
    }

    this.cibles = this.cibles.remove_elements(aEnlever);

    if (this.cibles.length == 0) {
        this.setEtat(navigator.Etat.DEPLACEMENT);
    }

    if (nbrFois > 0) {
        this.setVie(0);
    }
};

VeggieKamikaz.prototype.action = function(nbFois) {
    return null;
};
